api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -630,6 +630,13 @@ message HttpConnectionManager {
   // <envoy_api_msg_extensions.filters.network.http_connection_manager.v3.PathNormalizationOptions>`
   // for details.
   PathNormalizationOptions path_normalization_options = 43;
+
+  // Determines if trailing dot of the host should be removed from host/authority header before any
+  // processing of request by HTTP filters or routing.
+  // This affects the upstream host header.
+  // Without setting this option, incoming requests with host `example.com.` will not match against
+  // route with :ref:`domains<envoy_api_field_config.route.v3.VirtualHost.domains>` match set to `example.com`. Defaults to `false`.
+  bool strip_trailing_host_dot = 44;
```

Comment:
Drive by question for mdroth, do we think it's worth moving this and strip_matching_host_port together in a host mutation proto or just and wait and see if we get more?

Reply:
At this point, there are a whole bunch of options here related to URI normalization, not just the host ones but also the path ones.  I'd be open to moving all of those options to a separate `UriNormalizationOptions` message, but I'd also be okay with keeping them here, since they're unlikely to ever be used anywhere else.